app/components/designate-form.js

```diff
@@ -195,6 +195,44 @@ export default Ember.Component.extend({
     };
   }),
 
+  itemDesignateParams() {
```

Comment:
The chunk of code below is a bit mystical and unclear to me to be honest. Took me a significant amount of reads to understand 😄 

Few thoughts/questions

- where does this `cancelledState` come from ?

- You can probably refactor the url by code by doing `url = '/items/${id}/${action}'`

- Since the largest portion of the code is now in a huge param builder function, maybe creating methods which run the actions could be more understandable ? For clarity maybe using the API actions as function names could be good.
It would also allow us to add behaviours later on, as separate methods 

e.g
```javascript
isSameDesignation() { ... } // extract singular logics into methods

itemDesignateParams() { ... } // generic params

updatePartialQuantityOfDesignation() { ... } // action 1

designatePartialItem() { // action 2
   const url = this.buildUrl('designate_partial_item');
   const props = _.extend(this.itemDesignateParams(), { 
      ...
   });

  return ajax(...);
}

designateItem() { // main logic
  ...
  if (this.isSameDesignation()) {
     // We update the quantity
     return this.updatePartialQuantityOfDesignation();
  } else {
     // We re-designate to a new item
     return this.designatePartialItem();
  }
}
```

Those are just my subjective thoughts and opinions, I like to push for maintainable and easily upgradable code 😄 🚀